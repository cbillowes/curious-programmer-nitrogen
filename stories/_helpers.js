import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page