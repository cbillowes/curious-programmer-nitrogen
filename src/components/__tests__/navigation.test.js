import React from 'react'
import Navigation from '../navigation'
import { getSnapshot } from './_helpers'

describe(`Navigation`, () => {
  const toggleOnClick = jest.fn()

  it(`should render default navigation`, () => {
    const navigation = (
      <Navigation
        toggleOnClick={() => toggleOnClick}
      />
    )
    const tree = getSnapshot(navigation)
    expect(tree).toMatchSnapshot()
  })

  it(`should render open navigation`, () => {
    const navigation = (
      <Navigation
        toggleOnClick={() => toggleOnClick}
        isOpen={true}
      />
    )
    const tree = getSnapshot(navigation)
    expect(tree).toMatchSnapshot()
  })

  it(`should render navigation on path from window.location.href /tags`, () => {
    delete global.window.location
    global.window = Object.create(window)
    global.window.location = {
      pathname: `/tags`
    }

    const navigation = (
      <Navigation
        toggleOnClick={() => toggleOnClick}
        isOpen={true}
      />
    )
    const tree = getSnapshot(navigation)
    expect(tree).toMatchSnapshot()
  })

  it(`should render navigation on path /about`, () => {
    const navigation = (
      <Navigation
        toggleOnClick={() => toggleOnClick}
        isOpen={true}
        path="/about"
      />
    )
    const tree = getSnapshot(navigation)
    expect(tree).toMatchSnapshot()
  })
})
