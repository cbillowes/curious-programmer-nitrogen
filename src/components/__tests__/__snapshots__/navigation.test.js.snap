// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render default navigation 1`] = `
Array [
  <button
    className="menu-button"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-ellipsis-v fa-w-6 "
      data-icon="ellipsis-v"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>,
  <nav
    className="navigation closed"
  >
    <div
      className="menu"
    >
       
      <a
        href="/blog"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Blog
      </a>
       
       
      <a
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        About
      </a>
       
       
      <a
        href="/archives"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Archives
      </a>
       
       
      <a
        href="/tags"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Tags
      </a>
       
       
      <a
        href="/credits"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Credits
      </a>
       
       
      <a
        href="/privacy-policy"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Privacy Policy
      </a>
       
       
      <a
        href="/license"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        License
      </a>
       
       
      <a
        href="https://github.com/cbillowes"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        Github
      </a>
       
       
      <a
        href="https://www.linkedin.com/in/cbouwer/"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        LinkedIn
      </a>
       
    </div>
  </nav>,
]
`;

exports[`Navigation should render navigation on path /about 1`] = `
Array [
  <button
    className="menu-button active"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-ellipsis-v fa-w-6 "
      data-icon="ellipsis-v"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>,
  <nav
    className="navigation open"
  >
    <div
      className="menu"
    >
       
      <a
        href="/blog"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Blog
      </a>
       
       
      <a
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        About
      </a>
       
       
      <a
        href="/archives"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Archives
      </a>
       
       
      <a
        href="/tags"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Tags
      </a>
       
       
      <a
        href="/credits"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Credits
      </a>
       
       
      <a
        href="/privacy-policy"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Privacy Policy
      </a>
       
       
      <a
        href="/license"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        License
      </a>
       
       
      <a
        href="https://github.com/cbillowes"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        Github
      </a>
       
       
      <a
        href="https://www.linkedin.com/in/cbouwer/"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        LinkedIn
      </a>
       
    </div>
  </nav>,
]
`;

exports[`Navigation should render navigation on path from window.location.href /tags 1`] = `
Array [
  <button
    className="menu-button active"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-ellipsis-v fa-w-6 "
      data-icon="ellipsis-v"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>,
  <nav
    className="navigation open"
  >
    <div
      className="menu"
    >
       
      <a
        href="/blog"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Blog
      </a>
       
       
      <a
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        About
      </a>
       
       
      <a
        href="/archives"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Archives
      </a>
       
       
      <a
        href="/tags"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Tags
      </a>
       
       
      <a
        href="/credits"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Credits
      </a>
       
       
      <a
        href="/privacy-policy"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Privacy Policy
      </a>
       
       
      <a
        href="/license"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        License
      </a>
       
       
      <a
        href="https://github.com/cbillowes"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        Github
      </a>
       
       
      <a
        href="https://www.linkedin.com/in/cbouwer/"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        LinkedIn
      </a>
       
    </div>
  </nav>,
]
`;

exports[`Navigation should render open navigation 1`] = `
Array [
  <button
    className="menu-button active"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-ellipsis-v fa-w-6 "
      data-icon="ellipsis-v"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>,
  <nav
    className="navigation open"
  >
    <div
      className="menu"
    >
       
      <a
        href="/blog"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Blog
      </a>
       
       
      <a
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        About
      </a>
       
       
      <a
        href="/archives"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Archives
      </a>
       
       
      <a
        href="/tags"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Tags
      </a>
       
       
      <a
        href="/credits"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Credits
      </a>
       
       
      <a
        href="/privacy-policy"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Privacy Policy
      </a>
       
       
      <a
        href="/license"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        License
      </a>
       
       
      <a
        href="https://github.com/cbillowes"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        Github
      </a>
       
       
      <a
        href="https://www.linkedin.com/in/cbouwer/"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        LinkedIn
      </a>
       
    </div>
  </nav>,
]
`;
